wwwapp/forms.py

```diff
@@ -177,6 +177,11 @@ def __init__(self, *args, **kwargs):
         self.helper.form_tag = False
         self.helper.disable_csrf = True
         self.helper.include_media = False
+
+        year = self.instance.type.year if hasattr(self.instance, 'type') else settings.CURRENT_YEAR
+        self.fields['category'].queryset = WorkshopCategory.objects.filter(year=year)
```

Comment:
Remove queryset from lines 167 & 171

Reply:
You can't - it's a required constructor argument :/

Reply:
:'(